core/src/test/java/org/sonarsource/sonarlint/core/mediumtest/fixtures/ProjectStorageFixture.java

```diff
@@ -113,18 +118,27 @@ public RuleSetBuilder(String languageKey) {
       }
 
       public RuleSetBuilder withActiveRule(String ruleKey, String severity) {
-        activeRules.add(new ActiveRule(ruleKey, severity));
+        activeRules.add(new ActiveRule(ruleKey, severity, null, Map.of()));
+        return this;
+      }
+
+      public RuleSetBuilder withCustomActiveRule(String ruleKey, String templateKey, String severity, Map<String, String> params) {
+        activeRules.add(new ActiveRule(ruleKey, severity, templateKey, params));
         return this;
       }
     }
 
     private static class ActiveRule {
       private final String ruleKey;
       private final String severity;
+      private final String templateKey;
+      private final Map<String, String> params;
 
-      private ActiveRule(String ruleKey, String severity) {
+      private ActiveRule(String ruleKey, String severity, @Nullable String templateKey, Map<String, String> params) {
```

Comment:
Nitpick: I'd add shortcut constructors to avoid passing `null` and and empty map on 'simple' cases.

Reply:
There are builders to construct instances of this class so I don't think it is needed.